src/app/views/sidebar/resource-explorer/panels/PathsReview.tsx

```diff
@@ -21,37 +34,138 @@ export interface IPathsReview {
 
 const PathsReview = (props: IPathsReview) => {
   const dispatch: AppDispatch = useDispatch();
-  const { resources: { paths: items } } = useAppSelector(
-    (state) => state
-  );
+  const {
+    resources: { paths: items }
+  } = useAppSelector((state) => state);
   const { isOpen } = props;
-  const headerText = translateMessage('Selected Resources') + ' ' + translateMessage('Preview');
+  const headerText =
+    translateMessage('Selected Resources') + ' ' + translateMessage('Preview');
   const [selectedItems, setSelectedItems] = useState<IResourceLink[]>([]);
+  const [
+    showVersionErrorMessage,
+    {
+      toggle: toggleShowVersionErrorMessage,
+      setTrue: setShowVersionErrorMessageTrue,
+      setFalse: setShowVersionErrorMessageFalse
+    }
+  ] = useBoolean(false);
 
   const columns = [
-    { key: 'url', name: 'URL', fieldName: 'url', minWidth: 300, maxWidth: 350, isResizable: true }
+    {
+      key: 'url',
+      name: 'URL',
+      fieldName: 'url',
+      minWidth: 300,
+      maxWidth: 350,
+      isResizable: true
+    }
   ];
 
+  const getItemsForActions = () => {
+    if (selectedItems.length === 0) {
+      return items;
+    }
+    return selectedItems;
+  };
+
   const generateCollection = () => {
-    const content = generatePostmanCollection(items);
+    const content = generatePostmanCollection(getItemsForActions());
     const filename = `${content.info.name}-${content.info._postman_id}.postman_collection.json`;
     downloadToLocal(content, filename);
-  }
-
+  };
+  const getDescriptionUrl = () => {
+    const workingItems = getItemsForActions();
+    if (workingItems.length > 0 && workingItems[0].version === 'beta') {
+      return 'https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/openapi/beta/openapi.yaml';
+    }
+    return 'https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/openapi/v1.0/openapi.yaml';
+  };
+  const getVersionsCount = () => {
+    const workingItems = getItemsForActions();
+    const versions = workingItems.map((item) => item.version);
+    return new Set(versions).size;
+  };
+  const redirectToKiotaWeb = () => {
+    const includePatterns = getItemsForActions()
+      .map((item) => item.url)
+      .flat()
+      .join(',');
+    const versionCount = getVersionsCount();
+    if (versionCount > 1) {
+      setShowVersionErrorMessageTrue();
+    } else {
+      setShowVersionErrorMessageFalse();
+      window.open(
+        // eslint-disable-next-line max-len
+        `https://app.kiota.dev/generate?i=${includePatterns}&d=${getDescriptionUrl()}&b=True&ad=True&c=GraphServiceClient`,
+        '_blank'
+      );
+      trackGenerate();
```

Comment:
*Question:* Should we track the generation whether or not it was successful? Since the user has already demonstrated intent to generate

Reply:
Agreed, moved outside of the condition.